src/sys/signal.rs

```diff
@@ -831,6 +831,95 @@ mod sigevent {
     }
 }
 
+/// A wrapper for `stack_t`, used to represent alternate stacks for `sigaltstack`.
+#[derive(Copy, Clone)]
+#[allow(missing_debug_implementations)]
+pub struct SigStack {
+    stack: libc::stack_t,
+}
+
+impl SigStack {
+    /// Define an alternate signal stack for use with `sigaction`.
+    ///
+    /// See [`sigaltstack(3posix)`](http://man7.org/linux/man-pages/man3/sigaltstack.3p.html) for
+    /// details on the requirements for an alternate signal stack.
+    pub fn new(sp: *mut libc::c_void, flags: SigStackFlags, size: libc::size_t) -> SigStack {
```

Comment:
I really don't like this API.  It isn't Rusty.  For one thing, it exposes raw pointers in its public API.  For another, it forces the user to specify the size twice: here and wherever he allocated the memory.  Can you try to come up with a nicer API?  Taking the memory region as a mutable slice would be one option.  Handling allocation too would be even better, though we need to be careful not to proscribe valid use cases.

Reply:
I added an alternate interface that takes a mutable byte slice. Having `SigStack` itself own the allocation would increase complexity and decrease safety, particularly around how `sigaction`'s would be handled. Better for the user to know they have to explicitly manage unsafe lifetimes.